applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -2064,22 +2083,23 @@ GeometryData::IntegrationMethod SmallStrainUPwDiffOrderElement::GetIntegrationMe
     return GI_GAUSS;
 }
 
-void SmallStrainUPwDiffOrderElement::CalculateStrain(ElementVariables& rVariables, unsigned int GPoint)
+Vector SmallStrainUPwDiffOrderElement::CalculateStrain(const Matrix& rDeformationGradient,
+                                                       const Matrix& rB,
+                                                       const Vector& rDisplacements,
+                                                       bool          UseHenckyStrain)
 {
-    if (rVariables.UseHenckyStrain) {
-        rVariables.F             = this->CalculateDeformationGradient(GPoint);
-        rVariables.detF          = MathUtils<>::Det(rVariables.F);
+    if (UseHenckyStrain) {
         const SizeType Dim       = GetGeometry().WorkingSpaceDimension();
         const SizeType VoigtSize = (Dim == N_DIM_3D ? VOIGT_SIZE_3D : VOIGT_SIZE_2D_PLANE_STRAIN);
-        noalias(rVariables.StrainVector) = StressStrainUtilities::CalculateHenckyStrain(rVariables.F, VoigtSize);
+        return StressStrainUtilities::CalculateHenckyStrain(rDeformationGradient, VoigtSize);
     } else {
-        this->CalculateCauchyStrain(rVariables);
+        return this->CalculateCauchyStrain(rB, rDisplacements);
     }
 }
```

Comment:
As for the case of the small strain U-Pw element, I would remove the explicit `else` here.

Reply:
Done